workspace/sort-packages/src/index.ts

```diff
@@ -9,62 +9,10 @@ export function sequenceGraph (pkgGraph: ProjectsGraph): GraphSequencerResult<st
     keys.map((pkgPath) => [
       pkgPath,
       pkgGraph[pkgPath].dependencies.filter(
-        /* remove cycles of length 1 (ie., package 'a' depends on 'a').  They
-        confuse the graph-sequencer, but can be ignored when ordering packages
-        topologically.
-
-        See the following example where 'b' and 'c' depend on themselves:
-
-          graphSequencer({graph: new Map([
-            ['a', ['b', 'c']],
-            ['b', ['b']],
-            ['c', ['b', 'c']]]
-          ),
-          groups: [['a', 'b', 'c']]})
-
-        returns chunks:
-
-            [['b'],['a'],['c']]
-
-        But both 'b' and 'c' should be executed _before_ 'a', because 'a' depends on
-        them.  It works (and is considered 'safe' if we run:)
-
-          graphSequencer({graph: new Map([
-            ['a', ['b', 'c']],
-            ['b', []],
-            ['c', ['b']]]
-          ), groups: [['a', 'b', 'c']]})
-
-        returning:
-
-            [['b'], ['c'], ['a']]
-
-        */
-        d => d !== pkgPath &&
-        /* remove unused dependencies that we can ignore due to a filter expression.
-
-        Again, the graph sequencer used to behave weirdly in the following edge case:
-
-          graphSequencer({graph: new Map([
-            ['a', ['b', 'c']],
-            ['d', ['a']],
-            ['e', ['a', 'b', 'c']]]
-          ),
-          groups: [['a', 'e', 'e']]})
-
-        returns chunks:
-
-            [['d'],['a'],['e']]
-
-        But we really want 'a' to be executed first.
-        */
-        setOfKeys.has(d))]
+        d => d !== pkgPath && setOfKeys.has(d))]
```

Comment:
why were the comments removed?

Reply:
Because both scenarios are now supported

Reply:
Still retained ` d => d !== pkgPath && setOfKeys.has(d))]`, this is because it reduces the number of unnecessary cycles and handles external nodes. If the 'includeNodes' parameter contains nodes that are not within the graph, we cannot determine whether to discard them or treat them as separate isolated nodes. This decision should be made by the caller. If they should be discarded, they should be removed from 'includeNodes,' and if they are to be considered as standalone nodes within the graph, they should be added to the graph.